lms/static/js/spec/staff_debug_actions_spec.js

```diff
@@ -41,9 +49,45 @@ define(['backbone', 'jquery', 'js/staff_debug_actions'],
                 it('gets the placeholder name if the id has escapable characters', function() {
                     $('body').append(escapableFixture);
                     expect(StaffDebug.get_user('test.*+?^:${}()|][loc')).toBe('userman');
-                    $('#' + escapableFixture_id).remove();
+                    $("input[id^='sd_fu_']").remove();
                 });
             });
+            describe('do_idash_action success', function () {
+                it('adds a success message to the results DOM element', function () {
+                    $('body').append(escapableResultArea);
+                    var requests = AjaxHelpers.requests(this);
+                    var action = {
+                        locationName: esclocationName,
+                        location: location,
+                        method: 'reset_student_attempts',
+                        success_msg: gettext('Successfully reset the attempts for user userman'),
+                        error_msg: gettext('Failed to reset attempts.'),
+                        delete_module: false
+                    };
+                    StaffDebug.do_idash_action(action);
+                    AjaxHelpers.respondWithJson(requests, action);
+                    expect($('#idash_msg').text()).toBe('Successfully reset the attempts for user userman');
+                    $('#result_' + locationName).remove();
+                });
+            });
+            describe('do_idash_action error', function () {
+                it('adds a failure message to the results DOM element', function () {
+                    $('body').append(escapableResultArea);
+                    var requests = AjaxHelpers.requests(this);
+                    var action = {
+                        locationName: esclocationName,
+                        location: location,
+                        method: 'reset_student_attempts',
+                        success_msg: gettext('Successfully reset the attempts for user userman'),
+                        error_msg: gettext('Failed to reset attempts.'),
+                        delete_module: true
+                    };
+                    StaffDebug.do_idash_action(action);
+                    AjaxHelpers.respondWithError(requests);
+                    expect($('#idash_msg').text()).toBe('Failed to reset attempts. ');
```

Comment:
It is because of this line: https://github.com/edx/edx-platform/blob/vkaracic/TNL-394-confirmation-msg/lms/static/js/staff_debug_actions.js#L58
The test doesn't cause a JSON parsing error, hence the space is left without the 'Unknown Error Occurred.' addition.